examples/startup_namer/lib/main.dart

```diff
@@ -0,0 +1,109 @@
+import 'package:flutter/material.dart';
+import 'package:english_words/english_words.dart';
+
+void main() => runApp(new MyApp());
+
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      title: 'Startup Name Generator',
+      theme: new ThemeData(
+        primaryColor: Colors.white,
+      ),
+      home: new RandomWords(),
+    );
+  }
+}
+
+class RandomWords extends StatefulWidget {
+  @override
+  createState() => new RandomWordsState();
+}
+
+class RandomWordsState extends State<RandomWords> {
+  // Dart lets you define lists in different ways
+  final _suggestions = <WordPair>[];  
+  final List<WordPair> _saved = [];
+
+  final TextStyle _biggerFont = new TextStyle(fontSize: 18.0);
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+      appBar: new AppBar(
+        title: new Text('Startup Name Generator'),
+        actions: <Widget>[
+          new IconButton(
+            icon: new Icon(Icons.list),
+            onPressed: _pushSaved,
+          ),
+        ],
+      ),
+      body: _buildSuggestions(),
+    );
+  }
+
+  Widget _buildRow(WordPair pair) {
+    final alreadySaved = _saved.contains(pair);
+    return new ListTile(
+      title: new Text(
+        pair.asPascalCase,
+        style: _biggerFont,
+      ),
+      trailing: new Icon(
+        alreadySaved ? Icons.favorite : Icons.favorite_border,
+        color: alreadySaved ? Colors.red : null,
+      ),
+      onTap: () {
+        setState(() {
+          alreadySaved ? _saved.remove(pair) : _saved.add(pair);
+        });
+      },
+    );
+  }
+
+  Widget _buildSuggestions() {
+    return new ListView.builder(
+      padding: const EdgeInsets.all(16.0),
+      itemBuilder: (context, i) {
+        if (i.isOdd) return new Divider();
+
+        final index = i ~/ 2;
+        if (index >= _suggestions.length) {
+          _suggestions.addAll(generateWordPairs().take(10));
```

Comment:
you wouldn't want to setState here only because this is done in a build method, so by definition you're already building (setState is just a way to say "i need to rebuild). It's a bit weird to do work during build, but I don't know how else I'd implement this really. A Stream would be much more complicated (you'd have to pause it, get the data asynchronously, etc), and probably overkill since getting ten things from generateWordPairs is presumably cheap.

Reply:
Okay. I think we'll leave it like it is. Getting the random word pairs really is cheap. It's also not trying to emulate network requests at all. That will be for another tutorial. We're using pkg:english_words to a) teach about packages, b) give the starter app some interesting functionality.

> you wouldn't want to setState here only because this is done in a build method

Moreover, if I remember correctly, calling `setState` here will throw a run time error.